hermes/probe/monitor.go

```diff
@@ -12,9 +12,155 @@
+)
+
 // MonitorProbe holds aggregate information about all probe runs, per-target.
 // It also holds the config and options used to initialise the probe.
-type MonitorProbe struct{}
+type MonitorProbe struct {
+	name    string
+	config  *probepb.HermesProbeDef
+	targets []*probepb.Target
+	opts    *options.Options
+
+	// probeOpLatency is used to record latency values with distinct labels
+	// per probe operation per target.
+	// Recommended usage: append(probeOpLatency[target][probeOp], newMetric)
+	probeOpLatency map[string]map[string][]*metrics.EventMetrics
+	// apiCallLatency is used to record latency values with distinct labels
+	// per API call per target.
+	// Recommended usage: append(apiCallLatency[target][apiCall], newMetric)
+	apiCallLatency map[string]map[string][]*metrics.EventMetrics
+	logger         *logger.Logger
+}
+
+// Init initializes the probe with the given parameters.
+// This is a required method to implement the cloudprober.Probes.Probe interface.
+func (mp *MonitorProbe) Init(name string, opts *options.Options) error {
+	conf, ok := opts.ProbeConf.(*probepb.HermesProbeDef) // Casting ProbeConf interface to HermesProbeDef type
+	if !ok {
+		return fmt.Errorf("invalid argument: opts.ProbeConf is not of type *probepb.HermesProbeDef")
+	}
+	mp.name = name
+	mp.config = conf
+	mp.targets = mp.config.GetTargets()
+	mp.opts = opts
+	mp.probeOpLatency = make(map[string]map[string][]*metrics.EventMetrics)
+	mp.apiCallLatency = make(map[string]map[string][]*metrics.EventMetrics)
+	mp.logger = opts.Logger
+
+	return nil
+}
+
+// Start runs the probe indefinitely, unless cancelled, at the configured interval.
+// Probe metrics will be sent via the metricChan at the beginning of the next probe run.
+// Arguments:
+//	- ctx: context provided for cancelling probe.
+//	- metricChan: bidirectional channel used for sending metrics to be surfaced.
+func (mp *MonitorProbe) Start(ctx context.Context, metricChan chan *metrics.EventMetrics) {
+	probeTicker := time.NewTicker(mp.opts.Interval)
+
+	for {
+		select {
+		case <-ctx.Done():
+			probeTicker.Stop()
+			return
+		case <-probeTicker.C:
+			for _, target := range mp.probeOpLatency {
+				for _, op := range target {
+					for _, metric := range op {
+						metricChan <- metric
+					}
+				}
+			}
+
+			for _, target := range mp.apiCallLatency {
+				for _, call := range target {
+					for _, metric := range call {
+						metricChan <- metric
+					}
+				}
+			}
+
+			probeCtx, _ := context.WithDeadline(ctx, time.Now().Add(mp.opts.Timeout))
+			// init during probe run as metrics are timestamped.
+			if err := mp.initProbeMetrics(); err != nil {
+				mp.logger.Errorf("error initialising probe metric: %v", err)
+			}
+			mp.runProbe(probeCtx)
+		}
+	}
+}
+
+// initLatencyDistMetric initialises a latency distribution metric
+// according to the values passed.
+// Arguments:
+//	- target: pass target name.
+//	- latencyMap: pass a map with the key format: map[target][operation].
```

Comment:
Make it clear that this is the output.

Reply:
Ack.

Reply:
I don't see any change.
Please only mark a conversation resolved when the requested change was made.

Reply:
I did make a change to this in a commit on Friday:

I added: "-> This is the latency map being initialised with metrics and labels." beneath the argument comment.

However, I have made significant changes to the structure of metrics and removed this function since then.